vmhost/vmhooks/baseOps.go

```diff
@@ -2690,8 +2700,11 @@ func ExecuteReadOnlyWithTypedArguments(
 	metering.UseAndTraceGas(gasToUse)
 
 	sender := runtime.GetContextAddress()
+	originalCaller := runtime.GetOriginalCallerAddress()
+
 	contractCallInput, err := prepareIndirectContractCallInput(
 		host,
+		originalCaller,
```

Comment:
can't you have this inside the prepareIndirectContractCallInput ? from host you have access to runtime and GetOriginalCallerAddress

Reply:
fixed